### PR TITLE
[CHK-1566] feat(monitoring): use OpenTelemetry monitoring agent instead of Elastic APM

### PR DESCRIPTION
**Pull request by giovanniberti**
#### List of Changes

* use OpenTelemetry monitoring agent instead of Elastic APM

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: helm/values-dev.yaml [hunk 1]
```diff
@@ -79,14 +79,17 @@ microservice-chart:
...
     CARTS_MAX_ALLOWED_PAYMENT_NOTICES: "5"
-    ELASTIC_APM_SERVICE_NAME: pagopa-ecommerce-payment-requests-service
-    ELASTIC_APM_APPLICATION_PACKAGES: it.pagopa.ecommerce.payment.requests
-    ELASTIC_APM_SERVER_URL: http://quickstart-apm-http.elastic-system.svc:8200
-    ELASTIC_APM_LOG_SENDING: 'true'
+    OTEL_RESOURCE_ATTRIBUTES: "service.name=pagopa-ecommerce-payment-requests-service,service.version=0.2.2,deployment.environment=dev"
+    OTEL_EXPORTER_OTLP_ENDPOINT: "http://otel-collector.elastic-system.svc:4317"
+    OTEL_TRACES_EXPORTER: otlp
+    OTEL_METRICS_EXPORTER: otlp
+    OTEL_LOGS_EXPORTER: otlp
```
Comment (pietro-tota) [inline helm/values-dev.yaml:86]:
```suggestion
    OTEL_LOGS_EXPORTER: none
```

Log shipping should be done by filebeat, right?

File: helm/values-uat.yaml [hunk 2]
```diff
@@ -96,10 +96,12 @@ microservice-chart:
     NODO_READ_TIMEOUT: "10000"
     NODO_CONNECTION_TIMEOUT: "10000"
     CARTS_MAX_ALLOWED_PAYMENT_NOTICES: "5"
-    ELASTIC_APM_SERVICE_NAME: pagopa-ecommerce-payment-requests-service
-    ELASTIC_APM_APPLICATION_PACKAGES: it.pagopa.ecommerce.payment.requests
-    ELASTIC_APM_SERVER_URL: http://quickstart-apm-http.elastic-system.svc:8200
-    ELASTIC_APM_LOG_SENDING: "true"
+    OTEL_RESOURCE_ATTRIBUTES: "service.name=pagopa-ecommerce-payment-requests-service,deployment.environment=uat"
```
Comment (pietro-tota) [inline helm/values-uat.yaml:99]: Modify also prod helm values

Comment (giovanniberti) [reply helm/values-uat.yaml]: Solved in https://github.com/pagopa/pagopa-ecommerce-payment-requests-service/pull/61/commits/54cc0848ef669df342967edaf655dbacd0a6b790

File: helm/values-dev.yaml (hunk 1 above)
Comment (giovanniberti) [reply helm/values-dev.yaml]: Solved in https://github.com/pagopa/pagopa-ecommerce-payment-requests-service/pull/61/commits/54cc0848ef669df342967edaf655dbacd0a6b790

File: helm/values-dev.yaml [hunk 3]
```diff
@@ -79,16 +79,16 @@ microservice-chart:
     NODO_READ_TIMEOUT: "10000"
     NODO_CONNECTION_TIMEOUT: "10000"
     CARTS_MAX_ALLOWED_PAYMENT_NOTICES: "5"
-    ELASTIC_APM_SERVICE_NAME: pagopa-ecommerce-payment-requests-service
-    ELASTIC_APM_APPLICATION_PACKAGES: it.pagopa.ecommerce.payment.requests
-    ELASTIC_APM_SERVER_URL: http://quickstart-apm-http.elastic-system.svc:8200
-    ELASTIC_APM_LOG_SENDING: "false"
-    ECS_SERVICE_NAME: pagopa-ecommerce-payment-requests-service
-    ECS_SERVICE_ENVIRONMENT: "dev"
```
Comment (pietro-tota) [inline helm/values-dev.yaml:87]: restore this two env variables, they are used by logback encoder to encode those informations into json logs

Comment (giovanniberti) [reply helm/values-dev.yaml:87]: Ty :)

Comment (giovanniberti) [reply helm/values-dev.yaml]: Fixed in 8bf5fee

**Review by pietro-tota — APPROVED** [review]
🚀

**Review by infantesimone — APPROVED** [review]

**Pull request merged**